policyeval/evaluate.go

```diff
@@ -139,5 +139,22 @@ func (pe *PolicyEvaluator) ReevaluateAffectedByLists(ctx context.Context, policy
 }
 
 func (pe *PolicyEvaluator) ReevaluateActions(ctx context.Context, actions []*database.TakenAction) {
-
+	for _, action := range actions {
+		if action.TargetUser == "" {
+			zerolog.Ctx(ctx).Warn().Any("action", action).Msg("Action has no target user")
+			continue
+		}
+		// unban users that were previously banned by this rule
+		if action.ActionType == database.TakenActionTypeBanOrUnban && action.Action == event.PolicyRecommendationBan {
```

Comment:
This is still missing 2 checks:

1. whether the source list has auto_unban set
2. whether there are other bans matching the user (that's why it's called `ReevaluateActions`)